static/js/pages/instances/DownloadButton.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { FC, useRef } from "react";
+import { Button, Icon, Tooltip } from "@canonical/react-components";
+import { LxdInstance } from "types/instance";
+import { fetchInstanceLogFile } from "api/instances";
+
+interface DownloadButtonProps {
+  instance: LxdInstance;
+  fileName: string;
+  content: string;
+}
+
+const DownloadButton: FC<DownloadButtonProps> = ({
+  instance,
+  content,
+  fileName,
+}) => {
+  const buttonRef = useRef<HTMLDivElement | null>(null);
+
+  const generateLink = (linkId: string, href: string) => {
+    const link = document.createElement("a");
+    link.setAttribute("href", href);
+    link.setAttribute("target", "_blank");
+    link.setAttribute("download", fileName);
+    link.setAttribute("id", linkId);
+    link.style.display = "none";
+
+    return link;
+  };
+
+  const handleDownload = async () => {
+    const linkId = `link-${fileName.toLowerCase().replace(" ", "-")}`;
+    const maybeExistingLink = document.getElementById(linkId);
+
+    if (maybeExistingLink) {
+      maybeExistingLink.click();
+      return;
+    }
+
+    const data =
+      content ||
+      (await fetchInstanceLogFile(instance.name, instance.project, fileName));
+    const blob = new Blob([data], { type: "text/plan" });
```

Comment:
There is a typo in the MIME type, it should be `text/plain`.

Reply:
good catch!